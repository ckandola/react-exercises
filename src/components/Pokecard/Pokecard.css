.entry {
  background-color: silver;
  border-radius: 2em;
  max-width: fit-content;
  margin:auto;
  margin-bottom: 6em;
  padding-bottom: 10em;
}

.pokemon-name {
  color: blue;
  font-size: 1.3em;
  text-align: center;
  padding-left: 0.5em;
  padding-right: 0.5em;
  padding-top: 0.5em;
}

.type {
  text-align: center;
  padding:0.5em;
  padding-top: 0em;
  padding-bottom: 0em;
  color: navy;
  font-size: 1em;
}

.image {
  margin-left: auto;
  margin-right: auto;
  display: block;
}
