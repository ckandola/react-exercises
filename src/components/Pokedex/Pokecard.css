.pokedex-entry {
  background-color: silver;
  border-radius: 2em;
  width: 10em;
  margin:auto;
  margin-bottom: 2em;
  padding-bottom: 1em;
}

.pokemon-name {
  color: blue;
  font-size: 1.3em;
  text-align: center;
  padding-left: 0.5em;
  padding-right: 0.5em;
  padding-top: 0.5em;
}

.pokemon-type {
  text-align: center;
  padding:0.5em;
  padding-top: 0em;
  padding-bottom: 0em;
  color: navy;
  font-size: 1em;
}

.pokedex-image {
  margin-left: auto;
  margin-right: auto;
  display: block;
}
