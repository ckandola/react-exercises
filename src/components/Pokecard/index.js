import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Pokecard.css";

class Pokecard extends Component {
  render() {
    return (
      <li key={this.props.id}>
        <div className="entry">
          <h3 className="pokemon-name">{this.props.name}</h3>
          <h3 className="type">Type: {this.props.type}</h3>
          <img src={this.props.image} alt={this.props.name} className="image" />
        </div>
      </li>
    );
  }
}

export default Pokecard;

Pokecard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  image: PropTypes.string,
  type: PropTypes.string
};
