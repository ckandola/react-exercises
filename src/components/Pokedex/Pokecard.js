import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Pokecard.css";

class Pokecard extends Component {
  render() {
    return (
      <li key={this.props.id}>
        <div className="pokedex-entry">
          <h3 className="pokemon-name">{this.props.name}</h3>
          <h3 className="pokemon-type">Type: {this.props.type}</h3>
          <img src={this.props.image} alt={this.props.name} className="pokedex-image" />
        </div>
      </li>
    );
  }
}

export default Pokecard;

Pokecard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  image: PropTypes.string,
  type: PropTypes.string
};
