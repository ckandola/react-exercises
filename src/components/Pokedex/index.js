import React, { Component } from "react";
import Pokecard from "../Pokecard";
import "./Pokedex.css";

class Pokedex extends Component {
  render() {
    const pokecards = this.props.pokemonList.map(pokemon => {
      return (
        <Pokecard
          key={pokemon.id}
          id={pokemon.id}
          name={pokemon.name}
          image={pokemon.image}
          type={pokemon.type}
        />
      );
    });
    return (
      <div>
        <h1 className="title">Pok√©dex</h1>
        <ul>{pokecards}</ul>
      </div>
    );
  }
}

export default Pokedex;

Pokedex.defaultProps = {
  pokemonList: [
    {
      id: 1,
      name: "Charmander",
      type: "fire",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
    },
    {
      id: 2,
      name: "Squirtle",
      type: "water",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
    },
    {
      id: 3,
      name: "Butterfree",
      type: "flying",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
    },
    {
      id: 4,
      name: "Rattata",
      type: "normal",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/19.png"
    },
    {
      id: 5,
      name: "Metapod",
      type: "bug",
      image:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
    }
  ]
};
