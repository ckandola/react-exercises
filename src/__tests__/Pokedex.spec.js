import React from 'react';
import Pokedex from '../components/Pokedex';
import Pokecard from '../components/Pokedex/Pokecard';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Pokedex test', () => {
    it('matches the snapshot', () => {
        const pokedex = shallow(<Pokedex />);
        expect(toJson(pokedex)).toMatchSnapshot();
    });

    it('Pokecard renders properly', () => {
        const pokecard = shallow(<Pokecard 
            key={382} id={382} name="Kyogre"
            type="water" image={"http://www.psypokes.com/dex/picdex/bw/382.png"}
        />);
        expect(pokecard.find('.pokemon-name').getElement().props.children).toBe("Kyogre");
    });
});
