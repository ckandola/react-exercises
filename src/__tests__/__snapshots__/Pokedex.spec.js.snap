// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokedex test matches the snapshot 1`] = `
<div>
  <h1
    className="pokedex-title"
  >
    Pok√©dex
  </h1>
  <ul>
    <Pokecard
      id={1}
      image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
      key="1"
      name="Charmander"
      type="fire"
    />
    <Pokecard
      id={2}
      image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
      key="2"
      name="Squirtle"
      type="water"
    />
    <Pokecard
      id={3}
      image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
      key="3"
      name="Butterfree"
      type="flying"
    />
    <Pokecard
      id={4}
      image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/19.png"
      key="4"
      name="Rattata"
      type="normal"
    />
    <Pokecard
      id={5}
      image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
      key="5"
      name="Metapod"
      type="bug"
    />
  </ul>
</div>
`;
